### PR TITLE
Security improvements

### PR DESCRIPTION
**Pull request by Killerqu00**
A proposal with several security issues being put in a single design doc.
This doc proposes:
- Rework to tazers, adding them back to security;
- Improvements for alert code system;
- Gameplay implications for having high alert code set;
- Some other minor tweak suggestions (that influence other departments at times, that's why I think they should be in a design doc too).

### CODE REVIEW

**Comment by ninruB** [comment]
Security has plenty of ways to cut the distance between the pursued and them, flashes, flashbangs, bolas (which are effectively what you propose for the tazer rework), you also naturally shorten the distance because of how airlocks function and how the station is close-quarters, and more situational or chance-based methods.

I don't see a reason to reintroduce tazers back to sec, especially as they are presented in this doc.

**Comment by Killerqu00** [comment]
> Security has plenty of ways to cut the distance between the pursued and them, flashes, flashbangs, bolas (which are effectively what you propose for the tazer rework), you also naturally shorten the distance because of how airlocks function and how the station is close-quarters, and more situational or chance-based methods.
> 
> I don't see a reason to reintroduce tazers back to sec, especially as they are presented in this doc.

i decided to cut out more deep analysis on the chase part, but here's some points on this:
- Airlocks work in favor of escapee, since escapee can hide behind airlocks with access;
- Flashes were made useless due to welding masks being pretty easy to obtain. However, this might change if masks get reduced FoV;
- Tazers in the form that was proposed in my doc were already PR'd and (IMHO) close to being approved, but author decided to close the PR.

**Comment by ninruB** [comment]
> * Airlocks work in favor of escapee, since escapee can hide behind airlocks with access;

Heavily depends on the exact chase situation, and typically the secoff will have enough time to squeeze in through the airlock, the secoff might have access or the pursued might NOT have access, maps like marathon have security checkpoints leading into department areas everywhere.

Airlocks passively slow the pursued down because they do not open instantly and most players bump into them rather than in-world interact from the further-most point of range.

> * Flashes were made useless due to welding masks being pretty easy to obtain. However, this might change if masks get reduced FoV;

Flashes are absolutely not obsoleted because some 1-5 people on-station (out of 80) choose to wear weldermasks, they still are extremely strong to the degree of being borderline overpowered.

Now I'll note that disablers are still very viable even on high ping even just to slow the pursued down, if not outright stamcrit them. I, personally, as a high ping (200-300ms) sec main, do not see any issue with using them.

Stinger grenades exist, just thought I should point this out as a FTR.

Neither do I think that chases should be trivialized, it's good if the suspect/pursued gets away because the secoff failed to catch them despite their toolkit being specialized for this; right now it's already very very well-oriented towards chases.

**Comment by Killerqu00** [comment]
> > * Airlocks work in favor of escapee, since escapee can hide behind airlocks with access;
> 
> Heavily depends on the exact chase situation, and typically the secoff will have enough time to squeeze in through the airlock, the secoff might have access or the pursued might NOT have access, maps like marathon have security checkpoints leading into department areas everywhere.
> 
> Airlocks passively slow the pursued down because they do not open instantly and most players bump into them rather than in-world interact from the further-most point of range.
> 
> > * Flashes were made useless due to welding masks being pretty easy to obtain. However, this might change if masks get reduced FoV;
> 
> Flashes are absolutely not obsoleted because some 1-5 people on-station (out of 80) choose to wear weldermasks, they still are extremely strong to the degree of being borderline overpowered.
> 
> Now I'll note that disablers are still very viable even on high ping even just to slow the pursued down, if not outright stamcrit them. I, personally, as a high ping (200-300ms) sec main, do not see any issue with using them.
> 
> Stinger grenades exist, just thought I should point this out as a FTR.
> 
> Neither do I think that chases should be trivialized, it's good if the suspect/pursued gets away because the secoff failed to catch them despite their toolkit being specialized for this; right now it's already very very well-oriented towards chases.

I can agree with flashes and partially with disablers, but airlocks are being used exactly like that.

Anyway, it's not the most important part. Other changes in the doc touch on airlock access issues and security checkpoints, too.

**Comment by thebadman4662** [comment]
Some kind of bola launcher smaller than 4x4 pneumatic cannon for sure would be nice, throws do not inherit momentum so ironically bola best use is not for chase at all but ambush or debuffing/stunning enemy that holds their ground or pursues you.

**Comment by ninruB** [comment]
I don't think this ultimately solves any "real" security issues. The forum/discord threads you've mentioned barely touch the design points you address in this doc.

* Message interruption as proposed is neat, but I'm pretty sure that's how it works right now.
* Harm mode indicators don't solve any issue as nothing stops Urist from turning on combat mode 0.3 seconds before hitting the secoff or vice-versa.
* Holographic projectors as proposed are, again, neat, but could use some downside for the user like having a slow upon trying to move through them.
* I don't think that handing out access based on the alert level is a good idea. This takes out even more interaction between heads, crew and sec.
Ideally you'd separate mass emergency access from alerts and give it to its own separate console in the bridge or something alike that.

**Comment by Killerqu00** [comment]
>     * Message interruption as proposed is neat, but I'm pretty sure that's how it works right now.

I thought so too, but apparently not, or at least it is inconsistent between different instances of damage, because I tested that with throwing knives.

>     * I don't think that handing out access based on the alert level is a good idea. This takes out even more interaction between heads, crew and sec.
>       Ideally you'd separate mass emergency access from alerts and give it to its own separate console in the bridge or something alike that.

I don't think that security constantly banging on the door (for example) of cargo while there is a dragon in there is "taking out interactions". They _could_ call a captain, HoP or QM, but they might be dead. In general, I think that this is just a redundant thing and should be migitated by having sec get soft AA on red. It kinda makes sense for a person that has a job to protect a department to access it at least in case of emergency.

File: src/en/space-station-14/departments/security/proposals/security-improvements.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+For the sake of simplicity, I propose that on LRP, any emergency allows for searches or using high-grade security gear. On MRP, however, security would have to follow this table: if there are no emergencies allowing for random searches, security cannot do them.
+
+This change is actually pretty minor: command already announces what is going on when changing alerts currently. This change will allow for more context for latejoiners and for crew that didn't get the context of said alert level raise for some reason.
+
+## Alarm Consequences
+
+For the sake of modularity, this will be discussed separately from the Alert Codes rework, using the current alert system. 
+
+Currently, there are zero consequences for abusing alarm system. You can just set the alarm to blue upon any minor crime being committed in order to give security a random search pass.
```
Comment (deltanedas) [inline src/en/space-station-14/departments/security/proposals/security-improvements.md:100]: wont this just make the problem worse

1. hos stubs toe on the door entering the bridge
2. code blue for the next hour
3. sec gets free guns as well as free search pass

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: src/en/space-station-14/departments/security/proposals/security-improvements.md (hunk 1 above)
Comment (Killerqu00) [reply src/en/space-station-14/departments/security/proposals/security-improvements.md:100]: That's valid, I should probably add something else to blue alert

**Comment by Cojoke-dot** [comment]
meh, harm mode indicators is kinda... ehhh... people don't use harm mode for harming people most of the time(sspppiiinnnn), and if someone is about to attack you they are just going to toggle it on when they start, not when it would be meaningful.

Taser change... is... is that not the point of the disabler? Feels like there is no point in slowing someone if I also have a tool that slows and stuns them that also works on hardsuits.

The alert level changes feel like they don't make much sense gameplay-wise. There is no incentive to be on green, it's an active debuff to the station. Yellow is nice, I like that one. Red just feels like such a reason to rush red alert then never turn it off, "I got slipped, go red so I can have AA"

**Comment by Killerqu00** [comment]
draft to make some edits and additions

**Pull request closed without merging**